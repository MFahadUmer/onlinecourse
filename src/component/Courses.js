import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Courses = ({
  uniqueKey,
  title,
  index,
  total,
  image,
  price,
}) => (
  <div className="coursesContainer">
    <div className="courseContainerCourse">
      <img className="courseImage" src={image} alt={image} />
      <div className="userCourseDetailsContainer">
        <div className="userCourseDetailsDetails">
          <h2 className="userCoursesTitle">
            <Link className="courseTitleLink" to={`/couseDetails/${uniqueKey}`}>{title}</Link>
          </h2>
          <p className="userCoursesPrice">
            {`${price} $`}
          </p>
        </div>
        <div className="userCourseDetailsDetails">
          <div>
            <span className="fas fa-star starColor" />
            <span className="fas fa-star starColor" />
            <span className="fas fa-star starColor" />
            <span className="fas fa-star starColor" />
            <span className="fas fa-star" />
          </div>
          <p> Per Month </p>
        </div>
      </div>
      <div className="numbering">
        {`${index} / ${total}`}
      </div>
    </div>
  </div>
);

Courses.propTypes = {
  title: PropTypes.string,
  uniqueKey: PropTypes.number,
  index: PropTypes.number,
  total: PropTypes.number,
  image: PropTypes.string,
  price: PropTypes.string,
};

Courses.defaultProps = {
  title: '',
  uniqueKey: '',
  index: '',
  total: '',
  image: '',
  price: '',
};

export default Courses;
