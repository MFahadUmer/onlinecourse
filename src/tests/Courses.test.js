import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Courses from '../component/Courses';

describe('Test Suit', () => {
  const courseId = 1;
  const userId = 1;
  const title = 'React and Redux';
  const details = 'A Very Nice Course';
  const price = '100';
  const requirements = 'No Requirements';
  const difficulty = 'Beginners';
  const image = 'difficulty';

  it('should render title on course component', () => {
    render(
      <BrowserRouter>
        <Courses
          key={courseId}
          uniqueKey={courseId}
          userType="user"
          title={title}
          details={details}
          requirements={requirements}
          difficulty={difficulty}
          image={image}
          price={price}
          author={userId}
          index="1"
          total="1"
        />
      </BrowserRouter>,
    );
    expect(screen.getByText('React and Redux')).toBeTruthy();
  });

  it('should Match the Courses componenrender title on course componentt snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Courses
          key={courseId}
          uniqueKey={courseId}
          userType="user"
          title={title}
          details={details}
          requirements={requirements}
          difficulty={difficulty}
          image={image}
          price={price}
          author={userId}
          index="1"
          total="1"
        />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
