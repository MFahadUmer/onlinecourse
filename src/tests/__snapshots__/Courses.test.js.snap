// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suit should Match the Courses componenrender title on course componentt snapshot 1`] = `
<div
  className="coursesContainer"
>
  <div
    className="courseContainerCourse"
  >
    <img
      alt="difficulty"
      className="courseImage"
      src="difficulty"
    />
    <div
      className="userCourseDetailsContainer"
    >
      <div
        className="userCourseDetailsDetails"
      >
        <h2
          className="userCoursesTitle"
        >
          <a
            className="courseTitleLink"
            href="/couseDetails/1"
            onClick={[Function]}
          >
            React and Redux
          </a>
        </h2>
        <p
          className="userCoursesPrice"
        >
          100 $
        </p>
      </div>
      <div
        className="userCourseDetailsDetails"
      >
        <div>
          <span
            className="fas fa-star starColor"
          />
          <span
            className="fas fa-star starColor"
          />
          <span
            className="fas fa-star starColor"
          />
          <span
            className="fas fa-star starColor"
          />
          <span
            className="fas fa-star"
          />
        </div>
        <p>
           Per Month 
        </p>
      </div>
    </div>
    <div
      className="numbering"
    >
      1 / 1
    </div>
  </div>
</div>
`;
